typescript/utils/src/utils.ts

```diff
@@ -34,30 +34,44 @@ export function addressToBytes32(address: Address): string {
     .toLowerCase();
 }
 
+export function formatCallData(
+  destinationContract: ethers.Contract,
+  functionStr: string,
+  functionArgs: any[],
```

Comment:
might be nice to type these `functionArgs` from the `destinationContract.interface.methods[function]`

Reply:
How would you do that in this case?

Reply:
`functionArgs: Parameters<C[functionStr]>` and have `function formatCallData<C extends ethers.Contract>`

Reply:
`functionStr` can also be `keyof C` or something